aws-core/src/main/java/com/amplifyframework/auth/AWSAuthSessionInternal.kt

```diff
@@ -16,6 +16,7 @@ package com.amplifyframework.auth
 
 import com.amplifyframework.auth.result.AuthSessionResult
 
+@Deprecated("This class was released with public visibility, but is not intended to be consumed.")
```

Comment:
Would the deprecation message generate errors during normal build, or only if a customer consumes the deprecated symbols directly in their app code?

Reply:
I *think* its only if they directly consume, but I will confirm. We have build warnings right now that will show when building Amplify directly that we can choose to suppress or ignore ( there are quite a few files I'll need to add a suppress warning on some of these internally).

Reply:
Could add a new annotation specific to amplify library? something similar to [this](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/Annotations.kt)

Reply:
@sdhuka I decided against it since we shouldn't remove these implementations since they were once public. I don't want us to forget down the line that removal would be a breaking change because that comment block wasn't always there. Open to reconsidering options here though, especially for a class like this that has "Internal" in the name.